Fix some logic flaws in NetworkServices settings change

Tolriq opened the pull request:
- When changing the SETTING_SERVICES_ESALLINTERFACES, JSONRPCServer needs to be restarted to apply the change, just a start is not enough since it first check if it's running.
- When toggling SETTING_SERVICES_ESENABLED, JSONRPCServer was never stopped since the there's a return after EventServer and before reaching this part of the code.

cc @Montellese 

Edit : And with a question also for @Montellese Is there a way in OnSettingChanging to detect if the change is from the GUI or from JSON ?

Because right now the EventServer restart code shows a dialog even when changed from JSON and this sounds problematic no ? (Same for Webserver start error for example and surely other cases)

Tolriq committed:
Fix some logic flaws in NetworkServices settings change

- When toggling SETTING_SERVICES_ESENABLED, JSONRPCServer was never stopped since the there's a return after EventServer and before reaching this part of the code.

Montellese commented on xbmc/network/NetworkServices.cpp:
```diff
@@ -328,31 +328,37 @@ bool CNetworkServices::OnSettingChanging(const CSetting *setting)
+      result = StopEventServer(true, true);
+#endif // HAS_EVENT_SERVER
+#ifdef HAS_JSONRPC
+      result = result && StopJSONRPCServer(false);
```
&=

Montellese commented:
Unfortunately there's no way to know where a setting change comes from and it's a known issue. We'd need some kind of context information to be able to do this.

Could you split the changes into two commits so that it's clear which commit fixes which issue?

Tolriq committed:
Fix JSONRpc interface listening on setting change

- When changing the SETTING_SERVICES_ESALLINTERFACES, JSONRPCServer needs to be restarted to apply the change, just a start is not enough since it first
check if it's running.

DaveTBlake commented:
@Montellese seems happy with this minor fix. Any reason not to merge anyone?

Montellese commented:
jenkins build and merge

Montellese commented:
@MartijnKaijser: should this be backported?
